### PR TITLE
split callbacks.py into a package

### PR DESCRIPTION
**Pull request by tgolsson**
The callbacks.py file was getting a bit too large so this breaks it down into a package with dedicated files per domain.

Also adds some more docs and probably picks up some lints.

Also makes the import error for wandb a bit more helpful by chaining it with some info in case it's not imported and adds a test case for it.

### CODE REVIEW

**Comment by singhblom** [comment]
Are we still waiting for my review here? Should we get back to this?

**Comment by tgolsson** [comment]
Just haven't had time to revisit!

**Comment by tgolsson** [comment]
@singhblom Updated and fixed your comments!

File: emote/callbacks/logging.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+from emote.callback import Callback
+from emote.mixins.logging import LoggingMixin
+
+
+class TensorboardLogger(Callback):
+    """Logs the provided loggable callbacks to tensorboard."""
+
+    def __init__(
+        self,
+        callbacks: List[LoggingMixin],
```
Comment (singhblom) [inline emote/callbacks/logging.py:17]: Call this `loggables` or something?

File: emote/callbacks/logging.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        writer: SummaryWriter,
+        log_interval: int,
+        log_by_samples: bool = False,
+    ):
+        super().__init__(cycle=log_interval)
+        self._cbs = callbacks
```
Comment (singhblom) [inline emote/callbacks/logging.py:23]: And then say `self._logs = loggables` like in the terminal logger.

**Review by singhblom — APPROVED** [review]
lgtm but two minor nitpicks.

**Pull request merged**